lib/core/util.js

```diff
@@ -487,19 +487,41 @@ function isValidHTTPToken (characters) {
   return true
 }
 
-// Parsed accordingly to RFC 9110
-// https://www.rfc-editor.org/rfc/rfc9110#field.content-range
-function parseRangeHeader (range) {
-  if (range == null || range === '') return { start: 0, end: null, size: null }
-
-  const m = range ? range.match(/^bytes (\d+)-(\d+)\/(\d+)?$/) : null
-  return m
-    ? {
-        start: parseInt(m[1]),
-        end: m[2] ? parseInt(m[2]) : null,
-        size: m[3] ? parseInt(m[3]) : null
-      }
-    : null
+const emptyRange = Object.freeze({ start: 0, end: null, size: null })
+const rangeHeaderRE = /^bytes (?:(?:(?:(\d+)-(\d+)))\/(?:\*|(\d+))$|\*\/(\d+))$/
+
+/**
+ * @typedef {Object} SatisfiedContentRangeHeader
+ * @property {number} start
+ * @property {number} end
+ * @property {number | null} size
+ */
+
+/**
+ * @typedef {Object} UnsatisfiedContentRangeHeader
+ * @property {0} start
+ * @property {null} end
+ * @property {number} size
+ */
+
+/**
+ * Parsed accordingly to RFC 9110
+ *
+ * @param {null|string} range
+ * @returns {null|SatisfiedContentRangeHeader|UnsatisfiedContentRangeHeader}
+ * @see https://www.rfc-editor.org/rfc/rfc9110#field.content-range
+ */
+function parseContentRangeHeader (range) {
+  if (range == null || range === '') {
```

Comment:
a header value can't be empty and I don't think range can be null here, only undefined